wolfssh/wolfsentry_test.h

```diff
@@ -0,0 +1,538 @@
+    struct wolfsentry_context *sentry,
+    const struct wolfsentry_action *action,
+    void *handler_arg,
+    void *caller_arg,
+    const struct wolfsentry_event *trigger_event,
+    wolfsentry_action_type_t action_type,
+    const struct wolfsentry_route *target_route,
+    struct wolfsentry_route_table *route_table,
+    struct wolfsentry_route *rule_route,
+    wolfsentry_action_res_t *action_results)
+{
+    const struct wolfsentry_event *parent_event = 
+                                    wolfsentry_route_parent_event(rule_route);
+    const char *tag_success = "authentication_succeeded";
+
+    (void)handler_arg;
+    (void)route_table;
+    (void)action_results;
+    (void)target_route;
+    (void)action_type;
+
+    fprintf(stderr, "action callback: a=\"%s\" parent_event=\"%s\" "
+                                "trigger=\"%s\" t=%u r_id=%u caller_arg=%p\n",
+            wolfsentry_action_get_label(action),
+            wolfsentry_event_get_label(parent_event),
+            wolfsentry_event_get_label(trigger_event),
+            action_type,
+            wolfsentry_get_object_id(rule_route),
+            caller_arg);
+
+    /* clear derogatory count */
+    if (XSTRNCMP(wolfsentry_event_get_label(trigger_event), tag_success,
+        XSTRLEN(tag_success)) ==0 && 
+        ((struct wolfsentry_data *)caller_arg)->rule_route) {
+            printf("clear derogatory count\n");
+            /* clear derogatory count */
+            wolfsentry_route_reset_derogatory_count(
+                    sentry, 
+                    ((struct wolfsentry_data *)caller_arg)->rule_route, NULL);
+            /* release object reference */
+             wolfsentry_route_drop_reference(sentry,
+                    ((struct wolfsentry_data *)caller_arg)->rule_route, NULL);
+            
+            ((struct wolfsentry_data *)caller_arg)->rule_route = NULL;
+    }
+    
+    if (rule_route) {
+
+        fprintf(stderr, "Route detail: \n");
+        wolfsentry_route_render(wolfsentry, rule_route, stdout);
+        
+        struct wolfsentry_route_metadata_exports metadata;
+        wolfsentry_route_get_metadata(rule_route, &metadata);
+
+        fprintf(stderr, " insert_time \t\t");  PRINT_DATE(sentry,
+                                                metadata.insert_time);
+        fprintf(stderr, " last_hit_time \t\t"); PRINT_DATE(sentry,
+                                               metadata.last_hit_time);
+        fprintf(stderr, " last_penaltybox_time \t"); PRINT_DATE(sentry,
+                                                metadata.last_penaltybox_time);
+        fprintf(stderr, " connection_count \t%d\n", metadata.connection_count);
+        fprintf(stderr, " derogatory_count \t%d\n", metadata.derogatory_count);
+        fprintf(stderr, " commendable_count \t%d\n", metadata.commendable_count);
+        fprintf(stderr, "\n");
+    }
+
+    return 0;
+}
+
+/*
+ * Callback that is fired when an action is taken excepts auth_success and auth_fial
```

Comment:
This line is over 80 characters, also needs spelling correction for "auth_fial".  Should be "auth_fail"?

Reply:
thanks. fixed